.github/workflows/ci.yml

```diff
@@ -40,3 +40,32 @@ jobs:
 
       - name: Run tests
         run: npm test
+
+  coverage:
```

Comment:
Instead of adding this huge `coverage` job block, we could try to use [jobs.<job_id>.steps.if](https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsif)  
Then after the `Run tests` step, we add the `Run coverage` and `Upload coverage to Codecov` steps with an `if: ${{ matrix.os == 'ubuntu-latest' && matrix.node == '12.x' }}`

Reply:
Nice, I didn't think about that. Let me do the change.